bambi/models.py

```diff
@@ -688,6 +688,74 @@ def posterior_predictive(
         else:
             return idata
 
+    # pylint: disable=protected-access
+    def predict(self, idata, kind="mean", scale="response", data=None):
+        """Predict method for Bambi models
+
+        Obtains in-sample and out-sample predictions from a fitted Bambi model.
+
+
+        Parameters
+        ----------
+        idata : InfereceData
+            ``InfereceData`` with samples from the posterior distribution.
+        kind: str
+            Indicates the type of prediction required. Can be ``"mean"`` or ``"distribution"``. The
```

Comment:
We should return samples and let the users compute the mean/median etc if then want. 

What about reserving the name "mean" for the posterior mean function and "pps" for the posterior predictive distribution? (once we deprecate `model.posterior_predictive()`)

Also by default we could have an `inplace` argument to control if we add the results to the passed idata or return a new one. If `kind=mean`, the results will be added to a new variable in the posterior group, we can call it "y_hat" by default or something similar (and/or let the user pass a name). And if `kind=pps` then the samples will be added to the `posterior_predictive` group for in-sample predictions and the `prediction` group for out-of-sample predictions.

Reply:
My comments:

* I think `"y_hat"` is more associated with the `"pps"` concept, because the response variable is what is being predicted.
* `kind="mean"` and `kind="pps"` are good choices. Let's discard `kind="distribution"`.
* Maybe `scale="linpred"` is clearer than `scale="link"`? Do you think there's any drawback with having another scale than the response? I'm thinking that `kind="pps"` only makes sense with`scale="response"`. An error could be raised when `kind="pps"` and `scale`  is not `"response"`, but I would like to know if you think this is confusing.
* If `kind="mean"` I agree we can add a new variable to the posterior.  If `scale="response"`, this can have the name of the response + "_mean", e.g. the response is `"vote"`, then the new variable is `"vote_mean"`. If `scale="linpred"` the name could be `"vote_linpred"`.

Finally, I'm good with returning an InferenceData object. But one of the things I like most about Bambi is that it is simple for newcomers. So I think we should have an example or section called **Recipes** or **How to** where we add several examples showing how to obtain predictions and create a variety of plots with them.

Reply:
Not strong opinion about `"y_hat"`. I agree is ambiguous I have seen used it for the mean of Y and for the pps of Y. 

I feel more inclined to remove the argument `scale` (but not opposed if you want to leave it). I guess most user will use the default and for the few user that may want  to do a transformation of the data they could do it manually, without too much trouble. We may want to expose the link functions to the user to make they live easier...

I agree with the rest, specially about more examples about how to operate with InferenceData. ArviZ has a Working with InferenceData section in the docs https://arviz-devs.github.io/arviz/getting_started/WorkingWithInferenceData.html (is still incomplete). We should both try to provide examples and report problems back to ArviZ to improve the usability of InferenceData.

Reply:
Good, I agree. I will do the following

* Remove `scale`.
* `Family` already has `link` (a string) and `_link` (a function) that converts the linear predictor to the mean. I'm going to modify it a little so we have `link_str` the name of the link function, `link` the link function itself, and `invlink` the inverse of the link function that goes from the mean to the linear predictor. Then we can have one or two examples where we show how to call `model.family.invlink()` to go from the mean prediction to the link scale. Sounds good?

Reply:
agreed

Reply:
I ended up creating a `Link` class. This has a `.name`, `.link`, `.linkinv` and `.linkinv_backend`. Built-in links are obtained just with a name. Also, I also clarify the distinction between `link` and `linkinv` so it is consistent with `g` and `g^-1` in GLMs

Reply:
This does not imply any change for the user, unless they want to create a custom ``Family`` with custom ``Link``, which would be very infrequent I think.